clients/src/main/java/org/apache/kafka/common/network/SslTransportLayer.java

```diff
@@ -159,10 +157,15 @@ public void close() throws IOException {
             }
             netWriteBuffer.flip();
             flush(netWriteBuffer);
-            socketChannel.socket().close();
-            socketChannel.close();
         } catch (IOException ie) {
             log.warn("Failed to send SSL Close message ", ie);
+        } finally {
+            try {
+                socketChannel.socket().close();
+                socketChannel.close();
+            } catch (IOException e) {
+                log.warn("Failed to close SSL socket channel: " + e);
```

Comment:
It was intentional since our logs are filled with exception stack traces from the exception "Failed to send SSL Close message" and the stack trace doesn't really add any value. I didn't want to remove the stack trace of the existing log, but thought it was unnecessary to add another one. I am happy to change it if you think it might be useful.